### ISSUE TITLE
Update minimum OS for production deployments

### ISSUE THREAD

**Issue by micheleRP** [redpanda-data/documentation#1700]
### Describe the Issue

- Update the variable "linux-kernel"  from "3.10.0-514 or 4.18" to "3.19 or 4.18"
- Remove RHEL/CentOS7 as a supported OS

### Updates to existing documentation

1. https://docs.redpanda.com/docs/deploy/deployment-option/self-hosted/kubernetes/kubernetes-cluster-requirements/
2. https://docs.redpanda.com/docs/deploy/deployment-option/self-hosted/manual/production/production-deployment/#prerequisites

**Comment by mattschumpert** [comment]
@micheleRP let's keep this on hold for now. we don't want to remove any supported OS

**Comment by Feediver1** [comment]
@emaxerrno FYI--Given that you [requested](https://redpandadata.slack.com/archives/C01ND4SVB6Z/p1686586076343839) that we make this min OS version update, I want to make you aware that PM has directed us to close the [PR ](https://github.com/redpanda-data/documentation/pull/1701) and to place this issue on hold.

**Comment by mattschumpert** [comment]
@emaxerrno , I hesitate to kill RHEL 7.9 support completely based on one anecdote when its fully supported by RedHat for at least another year, and defer to @jcsp on what we should do here, per him it was 3.10 absolute min Kernel, even if 3.19+ is 'strongly recommended'?

We certainly don't want to drop RHEL/CentOS support altogether!! (written above).  Basically a bunch of random comments in a slack thread by various folks should not change our OS support, we a core owner to drive this through PM.

**Comment by jcsp** [comment]
Kernel versions are awkward: Red Hat's "3.10" is actually a big pile of backports on top of upstream 3.10.

ScyllaDB do support RHEL 7, so I'd be surprised if the RHEL7 kernel was completely broken for AIO.

Redpanda doesn't currently test different OS versions, so there is no empirical basis for our claims about what does/doesn't work.

TLDR:
- If there is a business need to support RHEL7, then we should ask customers to open cases with Red Hat as/when they appear to hit kernel bugs.
- If there is not a business need to support RHEL7, then we should drop support for it.

**Comment by mattschumpert** [comment]
@jcsp, of course kernel bugs are the OS vendor's problem.  I don't want to drop support for anything if I can avoid it (clearly we see folks using RHEL 7.9)  but if there's a case to 'recommend' somethingm what is your take on saying anything (or nothing) about Min kernel version in docs as of now.   Should we eliminate it or say some kernal version or later is 'recommended'?  Similarly, RHEL 8+ is 'recommended' ?

**Comment by jcsp** [comment]
> Should we eliminate it or say some kernal version or later is 'recommended'? Similarly, RHEL 8+ is 'recommended' ?

Recommending that folks use RHEL >=8 is totally reasonable (as general life advice as well as for Redpanda :-))

It might indeed simplify matters if we avoid mentioning kernel versions, and just talk about distro versions.  People outside those distros aren't supported anyway, and if someone has issues on an ancient kernel, we can point them to the supported distro versions as an implicit baseline for oldest kernel versions.

**Comment by micheleRP** [comment]
@mattschumpert: How about updating [current doc](https://docs.redpanda.com/docs/deploy/deployment-option/self-hosted/manual/production/production-deployment/#hardware-and-software) to this: 

Follow these recommendations:

Operating system
- Minimum version of RHEL/CentOS: 8
- Minimum version of Ubuntu: 21.1

**Comment by mattschumpert** [comment]
@micheleRP let's remove the kernel version statement completely and just say ' Minimum recommended version for RHEL'. instead of 'Minimum version'
